From 3dabcf16f22a569957b4394b737ea134163301af Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 23 Nov 2017 02:24:28 +0000
Subject: [PATCH] operations/check: These are not the droids you are looking
 for

Actively skip `*.pyc` files that will ALWAYS change when run as root,
due to python "re-optimizing" them on install.

Actively skip module files that are affected by depmod, as these ALWAYS
change when installing NVIDIA drivers. Additionally `usysconf` will
enforce depmod changes for all module installations automatically, so
this will become an issue for all Solus users.

Basically, make `eopkg check` lie a little less and stop scaring people.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/operations/check.py | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/pisi/operations/check.py b/pisi/operations/check.py
index f9fec7e..467333f 100644
--- a/pisi/operations/check.py
+++ b/pisi/operations/check.py
@@ -30,6 +30,37 @@ def file_corrupted(pfile):
             raise e
     return False
 
+
+# These guys will change due to depmod calls.
+_blessed_kernel_borks = [
+    "modules.alias",
+    "modules.alias.bin",
+    "modules.dep",
+    "modules.dep.bin",
+    "modules.symbols",
+    "modules.symbols.bin",
+]
+
+
+def ignorance_is_bliss(f):
+    """ Too many complaints about things that are missing. """
+    p = f
+    if not p.startswith("/"):
+        p = "/{}".format(f)
+
+    pbas = os.path.basename(p)
+    p = p.replace("/lib64/", "/lib/")
+
+    # Ignore kernel depmod changes?
+    if p.startswith("/lib/modules"):
+        if pbas in _blessed_kernel_borks:
+            return True
+
+    # Running eopkg as root will mutate .pyc files. Ignore them.
+    if p.endswith(".pyc"):
+        return True
+
+
 def check_files(files, check_config=False):
     results = {
                 'missing'   :   [],
@@ -43,6 +74,8 @@ def check_files(files, check_config=False):
             continue
         if not f.hash:
             continue
+        if ignorance_is_bliss(f.path):
+            continue
 
         is_file_corrupted = False
 
-- 
2.15.0

